src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1185 @@
+        for base in (1..buf.len()).step_by(2) {
+            buf.swap(base, base - 1);
+        }
+    }
+    // ------------------------------------------------------------------
+    // Read
+    let mut buf = vec![BUF_INIT_BYTE; bsize];
+    let mut rstat = match i.cflags.sync {
+        Some(ch) => i.fill_blocks(&mut buf, ch)?,
+        _ => i.fill_consecutive(&mut buf)?,
+    };
+    // Return early if no data
+    if rstat.reads_complete == 0 && rstat.reads_partial == 0 {
+        return Ok((rstat, buf));
+    }
+
+    // Perform any conv=x[,x...] options
+    if i.cflags.swab {
+        perform_swab(&mut buf);
+    }
+    if is_conv(i) || is_block(i) || is_unblock(i) {
+        let buf = conv_block_unblock_helper(buf, i, &mut rstat)?;
+        Ok((rstat, buf))
+    } else {
+        Ok((rstat, buf))
+    }
+}
+
+fn print_io_lines(update: &ProgUpdate) {
+    eprintln!(
+        "{}+{} records in",
+        update.read_stat.reads_complete, update.read_stat.reads_partial
+    );
+    if update.read_stat.records_truncated > 0 {
+        eprintln!("{} truncated records", update.read_stat.records_truncated);
+    }
+    eprintln!(
+        "{}+{} records out",
+        update.write_stat.writes_complete, update.write_stat.writes_partial
+    );
+}
+fn make_prog_line(update: &ProgUpdate) -> String {
+    let btotal_metric = Byte::from_bytes(update.write_stat.bytes_total)
+        .get_appropriate_unit(false)
+        .format(0);
+    let btotal_bin = Byte::from_bytes(update.write_stat.bytes_total)
+        .get_appropriate_unit(true)
+        .format(0);
+    let safe_millis = cmp::max(1, update.duration.as_millis());
+    let transfer_rate = Byte::from_bytes(1000 * (update.write_stat.bytes_total / safe_millis))
+        .get_appropriate_unit(false)
+        .format(1);
+
+    format!(
+        "{} bytes ({}, {}) copied, {:.1} s, {}/s",
+        update.write_stat.bytes_total,
+        btotal_metric,
+        btotal_bin,
+        update.duration.as_secs_f64(),
+        transfer_rate
+    )
+}
+fn reprint_prog_line(update: &ProgUpdate) {
+    eprint!("\r{}", make_prog_line(update));
+}
+fn print_prog_line(update: &ProgUpdate) {
+    eprintln!("{}", make_prog_line(update));
+}
+fn print_transfer_stats(update: &ProgUpdate) {
+    print_io_lines(update);
+    print_prog_line(update);
+}
+
+/// Generate a progress updater that tracks progress, receives updates, and responds to progress update requests (signals).
+fn gen_prog_updater(rx: mpsc::Receiver<ProgUpdate>, print_level: Option<StatusLevel>) -> impl Fn() {
+    // --------------------------------------------------------------
+    #[cfg(target_os = "linux")]
+    const SIGUSR1_USIZE: usize = signal::SIGUSR1 as usize;
+    // --------------------------------------------------------------
+    #[cfg(target_os = "linux")]
+    fn posixly_correct() -> bool {
+        env::var("POSIXLY_CORRECT").is_ok()
+    }
+    fn register_signal_handlers(sigval: Arc<AtomicUsize>) -> Result<(), Box<dyn Error>> {
```

Comment:
So this warns about `sigval` being unused on win/macos. To avoid this, I'd mark the whole function definition `#[cfg(target_os = "linux")]`, as well as the function call.

Reply:
I'm hoping to have `register_signal_handlers(...)` perform the work to register handlers for all platforms. To allow this extension, I think `#[allow(unused_variables)]` would be a better choice. I've added a comment to this effect in the file. 

We can also go a different route with this if you prefer.

Reply:
No, do as you prefer